tests/models/test_restore.py

```diff
@@ -199,9 +199,7 @@ def on_train_start(self):
             if self.trainer.state.fn == TrainerFn.TUNING:
                 self._test_on_val_test_predict_tune_start()
             else:
-                # `-1` because this checkpoint is saved `on_train_epoch_end` which is considered part of the epoch so
```

Comment:
so this is no longer true? since this was one of the key changes in your old PR.

Reply:
Checkpoints saved `on_train_epoch_end` are still considered part of the epoch and hence have its `epoch` value off by 1

The difference now is that we don't increase `trainer.current_epoch` directly on restart so we don't need this `-1` anymore. The comment does not apply here even though it's still true.